Source/JavaScriptCore/assembler/MacroAssemblerARM64.h

```diff
@@ -4102,19 +4102,9 @@ class MacroAssemblerARM64 : public AbstractMacroAssembler<Assembler> {
 
         // This is a signed multiple of two 32-bit values, producing a 64-bit result.
         m_assembler.smull(dest, src1, src2);
-        // Copy bits 63..32 of the result to bits 31..0 of scratch1.
-        m_assembler.asr<64>(scratch1, dest, 32);
-        // Splat bit 31 of the result to bits 31..0 of scratch2.
-        m_assembler.asr<32>(scratch2, dest, 31);
-        // After a mul32 the top 32 bits of the register should be clear.
+        m_assembler.cmp<64>(dest, dest, Assembler::SXTW, 0);
         zeroExtend32ToWord(dest, dest);
```

Comment:
We will remove this later by adding different method additionally in the next patch.